diff --git a/test/CMakeLists.txt b/test/CMakeLists.txt
index 4ed1403f..8673ed00 100644
--- a/test/CMakeLists.txt
+++ b/test/CMakeLists.txt
@@ -10,9 +10,9 @@ if(CMAKE_CXX_COMPILER_ID MATCHES "Clang")
 		message("GLM: Clang - ${CMAKE_CXX_COMPILER_ID} compiler")
 	endif()
 
-	if(NOT GLM_DISABLE_AUTO_DETECTION)
-		add_compile_options(-Werror -Weverything)
-	endif()
+	# if(NOT GLM_DISABLE_AUTO_DETECTION)
+	# 	add_compile_options(-Werror -Weverything)
+	# endif()
 
 elseif(CMAKE_CXX_COMPILER_ID MATCHES "GNU")
 	if(NOT GLM_QUIET)
@@ -20,7 +20,7 @@ elseif(CMAKE_CXX_COMPILER_ID MATCHES "GNU")
 	endif()
 
 	if(NOT GLM_DISABLE_AUTO_DETECTION)
-		add_compile_options(-Werror)
+#		add_compile_options(-Werror)
 #		add_compile_options(-Wpedantic)
 #		add_compile_options(-Wall)
 #		add_compile_options(-Wextra)
